patroni/ha.py

```diff
@@ -41,6 +43,54 @@ def demote(self):
     def follow_the_leader(self):
         return self.state_handler.follow_the_leader(self.cluster.leader)
 
+    @staticmethod
+    def fetch_node_status(member):
+        """This function perform http get request on member.api_url and fetches its status
+        :returns: tuple(`member`, reachable, in_recovery, xlog_location)
+
+        reachable - `!False` if the node is not reachable or is not responding with correct JSON
+        in_recovery - `!True` if pg_is_in_recovery() == true
+        xlog_location - value of `replayed_location` or `location` from JSON, dependin on its role."""
+
+        try:
+            response = requests.get(member.api_url, timeout=2)
+            json = response.json()
+            logger.info('Got response from %s: %s', member.name, json)
+            in_recovery = json.get('role', 'slave') != 'master'
+            xlog_location = json.get(('replayed_location' if in_recovery else 'location'), 0)
```

Comment:
Found a bug actually...
location and replayed_location are not in root node, but in a 'xlog'.